src/main/scala/edu/ie3/mobsim/model/ElectricVehicle.scala

```diff
@@ -131,51 +90,58 @@ case class ElectricVehicle(
     */
   def copyWith(
       storedEnergy: ComparableQuantity[Energy],
-      destinationPoiType: PoiTypeDictionary.Value,
-      destinationCategoricalLocation: CategoricalLocationDictionary.Value,
       destinationPoi: PointOfInterest,
       parkingTimeStart: ZonedDateTime,
       departureTime: ZonedDateTime
-  ): ElectricVehicle = {
-    this.storedEnergy = storedEnergy
-    this.destinationPoiType = destinationPoiType
-    this.destinationCategoricalLocation = destinationCategoricalLocation
-    this.destinationPoi = destinationPoi
-    this.parkingTimeStart = parkingTimeStart
-    this.departureTime = departureTime
-    this
-  }
+  ): ElectricVehicle =
+    copy(
+      storedEnergy = storedEnergy,
+      destinationPoi = destinationPoi,
+      parkingTimeStart = parkingTimeStart,
+      departureTime = departureTime
+    )
 
-  def setChosenChargingStation(chargingStation: Option[UUID]): Unit = {
-    chosenChargingStation = chargingStation
+  def setChosenChargingStation(
+      chargingStation: Option[UUID]
+  ): ElectricVehicle = {
+    copy(chosenChargingStation = chargingStation)
   }
 
-  def setChargingAtSimona(isCharging: Boolean): Unit = {
-    chargingAtSimona = isCharging
+  def setChargingAtSimona(): ElectricVehicle = {
+    copy(chargingAtSimona = true)
   }
 
-  def setFinalDestinationPoiType(
-      destinationPoiType: Option[PoiTypeDictionary.Value]
-  ): Unit = {
-    finalDestinationPoiType = destinationPoiType
+  def removeChargingAtSimona(): ElectricVehicle = {
+    copy(chargingAtSimona = false)
   }
 
-  def setFinalDestinationPoi(destinationPoi: Option[PointOfInterest]): Unit = {
-    finalDestinationPoi = destinationPoi
+  def setFinalDestinationPoi(
+      destinationPoi: Option[PointOfInterest]
+  ): ElectricVehicle = {
+    copy(finalDestinationPoi = destinationPoi)
   }
 
   def setRemainingDistanceAfterChargingHub(
       remainingDistance: Option[ComparableQuantity[Length]]
-  ): Unit = {
-    remainingDistanceAfterChargingHub = remainingDistance
+  ): ElectricVehicle = {
+    copy(remainingDistanceAfterChargingHub = remainingDistance)
   }
 
-  def updateChargingPricesMemory(newPrice: Double): mutable.Queue[Double] = {
+  def updateChargingPricesMemory(
```

Comment:
I like that you turned the `chargingPricesMemory` into an immutable queue. Please also only accept immutable queues as `newPrices` here. Once all data structures here have been made immutable, you can also just use `Queue[Double]` without the prefix.